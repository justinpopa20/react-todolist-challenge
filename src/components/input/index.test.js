import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Input from './'

describe('Input', () => {
  test('should render', () => {
    const wrapper = render(<Input />)

    expect(wrapper.baseElement).toMatchSnapshot()
  })

  test('should fire change event', () => {
    const onChange = jest.fn()

    const wrapper = render(<Input onChange={onChange} data-testid="input"/>)
    const input = wrapper.getByTestId("input")

    fireEvent.change(input, { target: { value: 'test123' } })    

    expect(onChange).toHaveBeenCalledWith('test123')
  })
})