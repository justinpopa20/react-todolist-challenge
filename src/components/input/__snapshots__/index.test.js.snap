// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<body>
  <div>
    <input
      type="text"
      value=""
    />
  </div>
</body>
`;
